cdap-docs/developers-manual/source/building-blocks/schedules.rst

```diff
@@ -23,8 +34,96 @@ Optionally, you can specify the properties for the Schedule::
+
+
+.. rubric:: Stream Size Schedules Lifecycle
+
+A **Stream Size Schedule** will execute a Workflow every time the Stream it references ingests an increment of data,
+also defined in the
+`Schedule. <../../reference-manual/javadocs/co/cask/cdap/api/schedule/Schedules.html#createDataSchedule(java.lang.String,%20java.lang.String,%20co.cask.cdap.api.schedule.Source,%20java.lang.String,%20int)>`__
+
+A few actions can be performed on a Schedule:
+
+- *Suspend*: suspending a schedule will prevent the Workflow from being executed again, in the event that the stream
+  ingests enough data. This action will not suspend the current execution of the Workflow, if the Schedule triggered
+  it recently. Suspending an already suspended Schedule has no effect;
+- *Resume*: this action will put the Schedule back into an active state, after a *Suspend* action. Resuming an already
+  active Schedule has no effect. If the Stream has ingested more than the increment of data defined by the Schedule
+  since the last time the Schedule was triggered, while the Schedule was suspended, then the Workflow will be executed
+  upon resume;
+- *Update*: this action is triggered when an application which contains a Stream Size Schedule is redeployed in CDAP.
+  If the Schedule name has not been modified and one of its parameter has changed - such as the `dataTrigger` - the
+  *Update* action will be called. If the `dataTrigger` is updated, and the Stream has ingested more data than defined
+  by the `dataTrigger` since the last time the Schedule was triggered, then the Workflow will be executed upon
+  update.
+
+.. rubric:: Stream Size Schedules Special Runtime Arguments
+
+When a **Stream Size Schedule** executes a Workflow, it passes to its context the following set of runtime arguments:
+
+- `logicalStartTime`: the timestamp, in milliseconds, at which the Schedule received the information that the Stream
+  had ingested enough data for the Workflow to be executed;
+- `runDataSize`: the size of the Stream, in bytes, seen by the Schedule when it triggered the current execution of
+  the Workflow;
+- `lastScheduledRunLogicalStartTime`: the `logicalStartTime` of the previous execution of the Workflow by this Schedule.
+  This argument is not present if the Workflow has never been executed by the Schedule;
+- `lastScheduledRunDataSize`: the `runDataSize` of the previous execution of the Workflow by this Schedule.
+  This argument is not present if the Workflow has never been executed by the Schedule;
+- `runBaseCountTime`: the timestamp, in milliseconds, at which the Schedule started to wait for an increment of data
+  being ingested by the Stream, before executing the Workflow. Most of the time, this argument and
+  `lastScheduledRunLogicalStartTime` will be identical, except during the first execution of the Workflow by the
+  Schedule, or if the counting logic in the Schedule is disrupted - for example by the deletion of Stream metrics;
```

Comment:
disrupted—for example, by the deletion of Stream metrics; \[add an em-dash and comma]